cleverhans/attacks_tf.py

```diff
@@ -1925,35 +1925,55 @@ def margin_logit_loss(model_logits, label, num_classes=10):
   return loss
 
 
-def _apply_transformation(x, dx, dy, angle, batch_size):
-  # Map a transformation onto the input
-  angle *= np.pi / 180
+def _apply_black_border(x, border_size):
+  orig_height = x.get_shape().as_list()[1]
+  orig_width = x.get_shape().as_list()[2]
+  x = tf.image.resize_images(x, (orig_width - 2*border_size,
+                                 orig_height - 2*border_size))
+
+  return tf.pad(x, [[0, 0],
+                    [border_size, border_size],
+                    [border_size, border_size],
+                    [0, 0]], 'CONSTANT')
+
+
+def _apply_transformation(inputs):
+  x, trans = inputs[0], inputs[1]
+  dx, dy, angle = trans[0], trans[1], trans[2]
   height = x.get_shape().as_list()[1]
   width = x.get_shape().as_list()[2]
-  M = np.array([1, 0, -dx*height,
-                0, 1, -dy*width, 0, 0] * batch_size, dtype=np.float32)
-  theta = tf.constant(M, shape=(batch_size, 8))
 
-  # Pad the image to prevent two-step rotation / translation
-  x = tf.pad(x, [[0, 0], [height, height], [width, width], [0, 0]],
+  # Pad the image to prevent two-step rotation / translation from truncating corners
+  max_dist_from_center = float(np.max([height, width])) * np.sqrt(2) / 2
```

Comment:
Probably doesn't matter, but what's wront with `np.sqrt(height**2+width**2) / 2`?

Reply:
Good point. That's more efficient.